internal/security/vaultclient/interface.go

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package vaultclient
+
+import (
+	"io"
+)
+
+// VaultClient is interface to Vault
+type VaultClient interface {
```

Comment:
can we try to not name this interface specific to vault, like

```suggestion
type SecretStoreClient interface {
```

Reply:
Changed as suggested.